jdaviz/utils.py

```diff
@@ -398,10 +398,13 @@ def flux_conversion(values, original_units, target_units, spec=None, eqv=None, s
             eqv = u.spectral_density(spectral_values)
         else:
             eqv = u.spectral_density(spec.spectral_axis[0])
-    elif slice is not None and eqv:
-        image_data = True
-        # Need this to convert Flux to Flux for complex conversions/translations of cube image data
-        eqv += u.spectral_density(slice)
+    elif spectral_axis and isinstance(spectral_axis, u.Quantity):
+        if eqv:
+            image_data = True
+            # Needed to convert Flux to Flux for complex conversion/translation of cube image data
+            eqv += u.spectral_density(spectral_axis)
+        elif len(values) == len(spectral_axis):
+            eqv = u.spectral_density(spectral_axis)
```

Comment:
@gibsongreen should there be an else here that raises an exception?  Can `spectral_axis` be a single value?

Reply:
I recall trying to avoid throwing exceptions here but in this case it probably makes sense to have it in the nested conditions. 

Originally slice was used for a single value, since Quantity objects can be both scalar or array-like generalizing this to check for both would be best.